controller/workitem_blackbox_test.go

```diff
@@ -2416,6 +2416,90 @@ func (s *WorkItem2Suite) TestDefaultSpaceAndIterationRelations() {
 	assert.Equal(s.T(), rootIteration.ID.String(), *wi.Data.Relationships.Iteration.Data.ID)
 }
 
+// Following test verifies that UPDATE on WI by setting AREA & Iteration
+// works as expected and do not alter previously set values
+func (s *WorkItem2Suite) TestWI2UpdateWithAreaIterationSuccessively() {
```

Comment:
Hi @aslakknutsen , There are two functions which test this use case.
[one](https://github.com/fabric8-services/fabric8-wit/blob/master/controller/workitem_blackbox_test.go#L1965) for the area 
[one](https://github.com/fabric8-services/fabric8-wit/blob/master/controller/workitem_blackbox_test.go#L2115) for iteration 
Those functions are getting passed with truth with my new code.
The things was -
```
u.Data.Relationships = &app.WorkItemRelationships{
		Iteration: &app.RelationGeneric{},
	}
```
this piece sets `Data=nil` as it is a pointer in `RelationGeneric`.